src/twisted/names/client.py

```diff
@@ -174,7 +174,10 @@ def parseConfig(self, resolvConf):
         for L in resolvConf:
             L = L.strip()
             if L.startswith(b'nameserver'):
-                resolver = (nativeString(L.split()[1]), dns.PORT)
+                parsedString = L.split()[1]
+                if b'.' in parsedString:
+                    parsedString = b'::' + parsedString
```

Comment:
That should work as expected, you can use a ipv6 socket to connect to an ipv4 address this way.